ENIGMAsystem/SHELL/Graphics_Systems/General/GSmodel.cpp

```diff
@@ -147,53 +147,50 @@ void string_parse(string *s) {
 namespace enigma_user {
 
 int d3d_model_create(int type, bool use_draw_color) {
-  int id = enigma::models.size();
-  enigma::Model* model = new enigma::Model(type, use_draw_color);
-  model->vertex_buffer = vertex_create_buffer();
-  enigma::models.push_back(model);
-  return id;
+  enigma::Model model(type, use_draw_color);
+  model.vertex_buffer = vertex_create_buffer();
+  return enigma::models.add(std::move(model));
 }
 
 void d3d_model_destroy(int id) {
-  const enigma::Model* model = enigma::models[id];
-  vertex_delete_buffer(model->vertex_buffer);
+  enigma::Model& model = enigma::models.get(id);
```

Comment:
You should be careful with your suggestions there. In fact, the reason I did not overload `operator[]` was because I didn't want it being abused. It's mostly never needed. Get is supposed to do bounds checking while `operator[]` does not.